### PR TITLE
validacija

### PR DESCRIPTION
**Pull request by Matej-Grubisic**
Napravljena validacija i ubaceno je provjeravanje jeli kriv unos i na invalidateu je ubaceno to da ako je vec invalidan giftcard da ga nece ope invalidira

### CODE REVIEW

File: docker/app/src/Controller/GiftcardController.php [hunk 1]
```diff
@@ -60,30 +62,57 @@ public function giftget($id): JsonResponse
...
-            'projectId' => self::projectId,
+            'projectId' => self::PROJECTID,
         ]);
 
+        $newNum = $request->toArray();
+
         $doc = $firestore->collection($_ENV['COLLECTION'])->document($id);
         $snapshot = $doc->snapshot();
 
+        $oldNum = $snapshot->data();
```
Comment (drapic88) [inline docker/app/src/Controller/GiftcardController.php]: $oldNum, $newNum naming convection is not a good practice. When you read the code, it should be clear what variables contain.

File: docker/app/src/Controller/GiftcardController.php [hunk 2]
```diff
@@ -60,30 +62,57 @@ public function giftget($id): JsonResponse
...
+        $oldNum = $oldNum['currency']['amount'];
+        
+        $newNum = $newNum['amount'];
+        
         if (!$snapshot->exists()) {
```
Comment (drapic88) [inline docker/app/src/Controller/GiftcardController.php:83]: Creating nest of if else is effective but pretty ugly to read and follow. You can do the same logic but much better.
For example, removing the else tag in this place. As you want that code to be executed if $snapshot exists, you have if condition for return the error message "This giftcard does not exist" and below the rest of the code.
Keep always in mind that code should be clean to read as possible. You will come back in some time on your code and you wont be able to read it :D

File: docker/app/src/Controller/GiftcardController.php [hunk 3]
```diff
@@ -93,30 +122,42 @@ public function giftredeem($id): JsonResponse
     /** 
      * @Route("/giftcard/invalidate/{id}", methods={"PATCH"})
      */
-    public function giftinvalidate($id): JsonResponse
+    public function giftInvalidate($id): JsonResponse
```
Comment (drapic88) [inline docker/app/src/Controller/GiftcardController.php:115]: for this method is the same comment as above.

**Comment by Matej-Grubisic** [comment]
Hvala, idem sada to popraviti!

uto, 26. srp 2022. u 08:44 drapic88 ***@***.***> napisao je:

**Review by drapic88 — APPROVED** [review]

**Pull request merged**